sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -76,6 +76,7 @@ private[sql] class SessionState(
     val sqlParser: ParserInterface,
     analyzerBuilder: () => Analyzer,
     optimizerBuilder: () => Optimizer,
+    statisticsCacheBuilder: () => StatisticsCache,
```

Comment:
This confuses me. Why do you maintain statistics here in the session state but not in a catalog, for instance.
Should be similar to the V1 command, I think:
https://github.com/apache/spark/blob/4a35a31c038f726f9329b4f28f3dde87286fb8d2/sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala#L242

Reply:
I am also a little unsure about this. Since V2Catalog does not have an interface for storing stats (it seems not suitable to add such an interface), stats can only be stored in SessionCatalog or Session. Is it appropriate to put the V2Table information in SessionCatalog? Or should put it into V2SessionCatalog (But I'm worried about breaking his role, which currently seems to be just a translation of SessionCatalog to V2Catalog)? Please give me some good suggestion. Thanks.

Reply:
сс @cloud-fan

Reply:
> Please give me some good suggestion. Thanks.

Our long term goal is to avoid special command for updating statistics. Like in Delta (https://github.com/delta-io/delta), gathering needed states is hidden from users. I would propose to output proper error message that V2 `ANALYZE TABLE` is not supported.

Reply:
Got it. Thanks for explain!